components/brave_wallet/browser/internal/sr25519.rs

```diff
@@ -45,24 +89,37 @@ fn generate_sr25519_keypair_from_seed(bytes: &[u8]) -> Box<CxxSchnorrkelKeyPair>
     // This function remains memory safe and if other C++ incorrectly invokes this
     // function, it will just simply panic which will mean it's easy to find.
     let mini_key = schnorrkel::MiniSecretKey::from_bytes(bytes).unwrap();
-    Box::new(CxxSchnorrkelKeyPair(mini_key.expand_to_keypair(schnorrkel::ExpansionMode::Ed25519)))
+    Box::new(CxxSchnorrkelKeyPair {
+        keypair: mini_key.expand_to_keypair(schnorrkel::ExpansionMode::Ed25519),
+        use_mock_rng: false,
+    })
 }
 
 impl CxxSchnorrkelKeyPair {
     fn get_public_key(self: &CxxSchnorrkelKeyPair) -> [u8; 32] {
-        self.0.public.to_bytes()
+        self.keypair.public.to_bytes()
+    }
+
+    fn use_mock_rng_for_testing(self: &mut CxxSchnorrkelKeyPair) {
+        self.use_mock_rng = true;
     }
 
     fn sign_message(self: &CxxSchnorrkelKeyPair, msg: &[u8]) -> [u8; 64] {
-        self.0.sign_simple(SIGNING_CTX, msg).to_bytes()
+        let t = schnorrkel::signing_context(SIGNING_CTX).bytes(msg);
+        if self.use_mock_rng {
+            let t = schnorrkel::context::attach_rng(t, MockRng(0));
```

Comment:
if we're only ever initializing `MockRng(0)`, may as well `#[derive(Default)]` and use `MockRng::default()`

Reply:
This no longer applies because we're using ChaCha20Rng from our vendored dependencies.